### ISSUE TITLE
Two problems in training code of ImageNetMini

### ISSUE THREAD

**Issue by ZhengyuZhao** [HanxunH/Unlearnable-Examples#7]
1. In line 173 of main.py,  20% of the PoisonImageNetMini were selected as the training data but with "shuffle=True", then how could you make sure these data are exactly the same 20% of the data that have been used for generating the noise?

2. I also find the same problem about the sample-wise noise as the one mentioned in https://github.com/HanxunH/Unlearnable-Examples/issues/5.
in line 634 of dataset.py:
```
if self.poison_samples[index]:
            noise = self.perturb_tensor[target]
            sample = sample + noise
            sample = np.clip(sample, 0, 255)
```
so by using 'target' as the index, the perturb_tensor[target] will only select one of the top '0~99' components of the perturb_tensor, and add the same noise to all samples that are from the same [target] class. In this way, it definitely does it wrong but can lead to good results because it is doing class-wise noise.

**Comment by HanxunH** [comment]
Hi,

For Q1, `np.random.seed(seed)` can make sure the randomly selected samples are the same. Make sure to use the same seed. 

For Q2, yes, that is an implementation error, `target` should be replaced with `index`, using `target` is class-wise noise. Need to make add an extra variable to distinguish the class-wise and sample-wise noise in this case, I will look into it.
